### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by PromzzyKoncepts**
#Hey There: 
> We have met all requirements for this project

- Created association relationships in Book and Classroom
- Created association relationships in Rentals and Person
- Created association relationships in Book and Rental

## Kindly review

### CODE REVIEW

**Review by youmari — COMMENTED** [review]
Hi @PromzzyKoncepts , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that 

- [ ] doesn't have a professional readme file please remove all the placeholders and fill all the sections with your information also don't delete the sections instead fill them out 
- [ ] it would be better to make sure that the `MIT` link is working and it should take you to the License of the project in order for your project will be professional

- [ ] please you should add a linter for ruby rubocop

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: students.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require './person'
+require './persons'
```
Comment (AnselemOdims) [inline students.rb:1]: - [ ] Good job working on this but please ensure that there is a method that serves as the 'belongs-to' `(a student belongs to a classroom)`.
- [ ] Also please kindly ensure that there is a getter for the classroom.
`attr_reader :classroom`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rental
```
Comment (AnselemOdims) [inline book.rb]: - [ ] Please kindly ensure that getters and setters for @rentals should be present

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (AnselemOdims) [inline rental.rb]: - [ ] Please ensure that getters and setters for @rentals should be present

File: persons.rb [hunk 4]
```diff
@@ -1,8 +1,7 @@
-require './nameable'
 require './base_decorator'
-require './capitalize_decorator'
+require './nameable'
 require './trimmer_decorator'
-
+require './capitalize_decorator'
 
 class Person < Nameable
   def initialize(name, age = 'Unknown', parent_permission: true)
```
Comment (AnselemOdims) [inline persons.rb:7]: - [ ] Please kindly ensure that there is a method for adding rentals
- [ ] Also, please ensure that getters and setters for @rentals, name, and age is present

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @PromzzyKoncepts , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by rustam-tolipov — APPROVED** [review]
Hi @PromzzyKoncepts, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**